File: python/Find Minimum in Rotated Sorted Array.py
class Solution(object):
    def findMin(self, nums):
        left = 0
        right = len(nums) - 1
        mid = (left + right) / 2
        while left <= right:
            if nums[left] <= nums[right]:
                return nums[left]
            mid = (left + right) / 2
            if (nums[mid] >= nums[left]):
                left = mid + 1
            else:
                right = mid
        
        return nums[mid]

File: cpp/Find Minimum in Rotated Sorted Array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int low = 0, high = nums.size() - 1;
        // while (low < high) {
        //     int mid = low + (high - low) / 2;
        //     if (nums[mid] < nums[low]) {
        //         high = mid;
        //     } else if (nums[mid] > nums[high]) {
        //         low = mid + 1;
        //     } else {
        //         high = mid - 1;
        //     }
        // }
        while (low < high) {
            int mid = low + (high - low) / 2;
            if (nums[mid] < nums[high]) {
                high = mid;
            } else {
                low = mid + 1;
            }
        }
        return nums[low];
    }
};
